examples/ui-demo/src/components/shared/MintCard.tsx

```diff
@@ -96,13 +107,19 @@ export const MintCard = () => {
       });
       console.log("uri", uri);
       return uri;
-
     },
     enabled: !!client && !!client?.readContract,
   });
 
   return (
     <div className="flex bg-bg-surface-default radius-1 border-btn-secondary border-2 overflow-hidden h-[532px]">
+      <Toast
```

Comment:
Do we plan on using this toast in multiple places. In the way it is currently implemented every component needs to instantiate and open + close their own toast, and handle the logic. 

Can we do better here? Could we wrap this all up in a toast provider component that exposes a useOpenToast function that takes in some parameters simply open and then auto close the toast globally? 

That way when we want to use it for errors, or something else, it's a bit easier for downstream components?

Reply:
looks like this is using a ToastProvider already, but it is weird that we can't just render a global Toast component and show a `Toast` via some kind of `showToast(...)` method. 

Maybe we could do something like this? https://www.radix-ui.com/primitives/docs/components/toast#imperative-api

the `useImperativeHandle` hook which contains a `publish` method is interesting here. I think we could use that to be able to:
1. render a single Toast instance in the root of the app
2. add to the context of the ui-demo app a ref or wrapper around a ref to the toast and export a method like `showToast({ content, type })`

Then we can have a `useToast` hook or something that lets us access that `showToast` method

Reply:
The intention was to have an explicit interface to reduce abstraction, I will refactor to use a hook implementation.